engine/access/ingestion2/collection_syncer.go

```diff
@@ -145,6 +171,24 @@ func (s *CollectionSyncer) StartWorkerLoop(ctx irrecoverable.SignalerContext, re
 			if err != nil {
 				ctx.Throw(err)
 			}
+
+		case <-s.pendingCollectionsHandler.GetNotifier():
+			msg, ok := s.pendingCollections.Get()
+			if !ok {
+				ctx.Throw(fmt.Errorf("could not get pending collection"))
+			}
+
+			collection, ok := msg.Payload.(*flow.Collection)
+			if !ok {
+				ctx.Throw(fmt.Errorf("could not cast pending collection to *flow.Collection"))
```

Comment:
let's include more detail so it's easier to debug if this ever comes up
```suggestion
				ctx.Throw(fmt.Errorf("could not cast pending collection to *flow.Collection. got: %T", msg.Payload))
```